tests/diffDataParent.js

```diff
@@ -0,0 +1,133 @@
+                javaRosa: {langs: ['en']},
+                core: {onReady: done}
+            });
+        });
+
+        it("should not allow repeat group children to have other data parents", function() {
+            util.loadXML("");
+            var repeat = util.addQuestion("Repeat", 'repeat1'),
+                text1 = util.addQuestion.bind({prevId: repeat.p.nodeID})("Text", 'text1'),
+                text2 = util.addQuestion.bind({prevId: repeat.p.nodeID})("Text", 'text2');
+
+            text2.p.dataParent = '/data';
+            assert(util.isTreeNodeValid(text1), "text1 should be valid");
+            assert(!util.isTreeNodeValid(text2), "text2 should not be valid");
+        });
+
+        it("should not allow a data parent to be an input", function() {
+            util.loadXML("");
+            var text1 = util.addQuestion("Text", 'text1'),
+                text2 = util.addQuestion("Text", 'text2');
+
+            text2.p.dataParent = '/data/text1';
+            assert(util.isTreeNodeValid(text1), "text1 should be valid");
+            assert(!util.isTreeNodeValid(text2), "text2 should not be valid");
+        });
+
+        it("should keep the same data parent after a move", function() {
+            util.loadXML("");
+            var text1 = util.addQuestion("Text", 'text1'),
+                text2 = util.addQuestion("Text", 'text2'),
+                form = call("getData").core.form;
+
+            util.addQuestion("Group", 'group1');
+            text2.p.dataParent = '/data/group1';
+
+            form.moveMug(text2, text1, 'before');
+
+            assert.equal(text2.p.dataParent, '/data/group1');
+        });
+
+        it("should update data tree after a change to data parent", function() {
+            util.loadXML("");
+            var text1 = util.addQuestion("Text", 'text1'),
+                group1 = util.addQuestion("Group", 'group1'),
+                group2 = util.addQuestion.bind({prevId: text1.p.nodeID})("Group", 'group2'),
+                form = call("getData").core.form;
+
+            text1.p.dataParent = '/data/group1';
+            assert.equal(text1.p.dataParent, "/data/group1");
+            form.moveMug(group1, group2, 'into');
+            assert.equal(text1.p.dataParent, "/data/group2/group1");
+        });
+
+        it("should clear the data parent when moving to a repeat group", function() {
+            util.loadXML("");
+            var text1 = util.addQuestion("Text", 'text1');
+            util.addQuestion("Group", 'group1');
+            var repeat1= util.addQuestion.bind({prevId: text1.p.nodeID})("Repeat", 'repeat1'),
+                form = call("getData").core.form;
+            text1.p.dataParent = '/data/group1';
+            form.moveMug(text1, repeat1, 'into');
+            assert(util.isTreeNodeValid(text1), "text1 should be valid");
+            assert.isUndefined(text1.p.dataParent);
+        });
+
+        it("should have proper data parent after change through UI", function() {
```

Comment:
I don't see where the UI is being changed in this test. Can you explain the name?

Possible suggested name: "should have proper data parent on reload form"